test/integration/puppetlabs/services/jruby/jruby_locking_test.clj

```diff
@@ -89,24 +96,34 @@
             (is (= [:instance-requested :instance-borrowed :instance-returned
                     :lock-requested :lock-acquired :lock-released] @events))))))))
 
-(deftest ^:integration borrow-and-return-affect-read-lock-test
-  (tk-bootstrap/with-app-with-config
-    app
-    [jruby-service/jruby-puppet-pooled-service
-     profiler/puppet-profiler-service]
-    (jruby-service-test-config 1)
-    (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
-          lock (jruby-testutils/get-lock-from-app app)]
-
-      (is (= 0 (.getReadLockCount lock)))
-      (is (not (.isWriteLocked lock)))
-
-      (let [instance (jruby-protocol/borrow-instance jruby-service :test-borrow-and-read-lock)]
-        (testing "borrow-instance acquires a read lock and does not affect write lock"
-          (is (= 1 (.getReadLockCount lock)))
-          (is (not (.isWriteLocked lock))))
-
-        (testing "return-instance releases a read lock and does not affect write lock"
-          (jruby-protocol/return-instance jruby-service instance :test-return-and-read-lock)
-          (is (= 0 (.getReadLockCount lock)))
-          (is (not (.isWriteLocked lock))))))))
+(deftest ^:integration with-lock-and-borrow-contention-test
+  (testing "contention for instances with borrows and locking handled properly"
+    (tk-bootstrap/with-app-with-config
+     app
+     [jruby-service/jruby-puppet-pooled-service
+      profiler/puppet-profiler-service]
+     (jruby-service-test-config 2)
+     (jruby-testutils/wait-for-jrubies app)
+     (let [jruby-service (tk-app/get-service app :JRubyPuppetService)]
+       (let [instance (jruby-protocol/borrow-instance
+                       jruby-service
+                       :with-lock-and-borrow-contention-test)]
+         (let [lock-thread-started? (promise)
```

Comment:
maybe rename this var to `lock-acquired?`?  technically the thread has been started before then.

/me ducks